android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java

```diff
@@ -1621,6 +1623,35 @@ public void setNestedScrollEnabled(boolean nestedScrollEnabled) {
       this.nestedScrollEnabled = nestedScrollEnabled;
     }
 
+    public void setSource(ReadableMap source) {
+      this.source = source;
+    }
+
+    public boolean isNewSource(ReadableMap newSource) {
+      if (source == null) {
+        return true;
+      }
+
+      // Check if any of the following string values have changed
+      String[] sourceKeys = {"uri", "method", "body", "html", "baseUrl"};
+
+      for (String key : sourceKeys) {
+        String value = source.getString(key);
+        String newValue = newSource.getString(key);
+        if (newValue != null && !newValue.equals(value)) {
+          return true;
+        }
+      }
+
+      // Check if headers changed
+      ReadableMap headersMap =  source.getMap("headers");
+      ReadableMap newHeadersMap = newSource.getMap("headers");
+      Map<String, Object> headers = headersMap == null ? Collections.emptyMap() : headersMap.toHashMap();
```

Comment:
In the for-loop a few lines above in this `isNewSource` method, it looks like we're assuming the values are Strings. Is it safe to make the type of `headers` and `newHeaders` `Map<String, String>` instead of `Map<String, Object>`?

Reply:
> headersMap.toHashMap()

returns a Map<String, Object> 

I can cast it so it so its' more explicit.

Reply:
With that context in mind, the code is clearer to me now as currently written. I think we can stick with the hashmap equals. Thanks for the context!